### PR TITLE
Fixed CMS deprecation warnings in SimPPS/RPDigiProducer

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

- handle esConsumes
- refactored to move EventSetup get to producer
- convert module to thread-friendly type

#### PR validation:

Packages compiles without issuing a CMS deprecation warning.

### CODE REVIEW

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
+1

File: SimPPS/RPDigiProducer/plugins/RPDetDigitizer.h [hunk 1]
```diff
@@ -2,6 +2,8 @@
 #define SimPPS_RPDigiProducer_RP_DET_DIGITIZER_H
 
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
```
Comment (perrotta) [inline SimPPS/RPDigiProducer/plugins/RPDetDigitizer.h:6]: not neede

File: SimPPS/RPDigiProducer/plugins/RPDetDigitizer.h [hunk 2]
```diff
@@ -2,6 +2,8 @@
 #define SimPPS_RPDigiProducer_RP_DET_DIGITIZER_H
 
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/EventSetup.h"
```
[the remark this answers is not in the document]
Comment (Dr15Jones) [reply SimPPS/RPDigiProducer/plugins/RPDetDigitizer.h]: It was a left over from my migration path.

**Comment by Dr15Jones** [comment]
please test

**Comment by perrotta** [comment]
please test

**Comment by civanch** [comment]
+1

File: SimPPS/RPDigiProducer/plugins/RPDisplacementGenerator.cc [hunk 3]
```diff
@@ -16,34 +12,37 @@
...
     S_m = ac.getTranslation();
     R_m = ac.getRotationMatrix();
   } else
     isOn_ = false;
 
   // transform shift and rotation to the local coordinate frame
-  ESHandle<CTPPSGeometry> geom;
-  iSetup.get<VeryForwardRealGeometryRecord>().get(geom);
-  const DetGeomDesc *g = geom->sensor(detId_);
+  //ESHandle<CTPPSGeometry> geom;
```
Comment (perrotta) [inline SimPPS/RPDigiProducer/plugins/RPDisplacementGenerator.cc:43]: Is this supposed to remain commented out? Or can we profit and remove these lines now?

Comment (Dr15Jones) [reply SimPPS/RPDigiProducer/plugins/RPDisplacementGenerator.cc]: done

**Comment by perrotta** [comment]
+1
- The last commit only removed some redundant commented out code: no need to bother @cms-sw/simulation-l2 to sign it once again

**Comment by perrotta** [comment]
merge

**Pull request merged**